### DISCUSSION TITLE
64-bit integer use of MUMPS always meet INFO(1)=-3

### DISCUSSION THREAD

**Discussion by GuoqiMa** [scivision/mumps-superbuild#51] [Q&A]
Hi, I am trying to using 64 bit integer of MUMPS. I add -i8 to the compiler as below,

_mpiifort -O2 -xHost -nofor-main -DBLR_MT -qopenmp -c -i8 $filename.f90 -o $filename.o

mpiifort -o $filename -O2 -xHost -nofor-main -qopenmp $filename.o -lzmumps -lmumps_common -lmpi -lpord -lmkl_scalapack_lp64 -lmkl_blacs_intelmpi_lp64 -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core -lpthread -lparmetis -lmetis -lptesmumps -lptscotch -lptscotcherr -lscotch_

But it doesn't help. always meet errors like this, I really need your help.  Do  the others like Metis, SCOTCH shoule be installed with 64-bit integer? Could you give some details? thanks very much.

 ** ERROR RETURN ** FROM ZMUMPS INFO(1)=   -3
 ** INFO(2)=               0
 ** ERROR RETURN ** FROM ZMUMPS INFO(1)=   -3
 ** INFO(2)=               0
 ** ERROR RETURN ** FROM ZMUMPS INFO(1)=   -3
 ** INFO(2)=               0
 ** ERROR RETURN ** FROM ZMUMPS INFO(1)=   -3
 ** INFO(2)=               0
 ** ERROR RETURN ** FROM ZMUMPS INFO(1)=   -3
 ** INFO(2)=               0
Abort(67706631) on node 2 (rank 2 in comm 0): Fatal error in PMPI_Bcast: Invalid root, error stack:
PMPI_Bcast(451): MPI_Bcast(buf=0x7fffdd81fd88, count=1, MPI_INTEGER, root=-1073741822, comm=comm=0x84000002) failed
PMPI_Bcast(412): Invalid root (value given was -1073741822)
Abort(134815495) on node 3 (rank 3 in comm 0): Fatal error in PMPI_Bcast: Invalid root, error stack:
PMPI_Bcast(451): MPI_Bcast(buf=0x7ffc093ad608, count=1, MPI_INTEGER, root=-1073741822, comm=comm=0x84000002) failed
PMPI_Bcast(412): Invalid root (value given was -1073741822)
Abort(671686407) on node 4 (rank 4 in comm 0): Fatal error in PMPI_Bcast: Invalid root, error stack:
PMPI_Bcast(451): MPI_Bcast(buf=0x7ffcf61d6808, count=1, MPI_INTEGER, root=-1073741822, comm=comm=0x84000002) failed
PMPI_Bcast(412): Invalid root (value given was -1073741822)
Abort(1007230727) on node 0 (rank 0 in comm 0): Fatal error in PMPI_Bcast: Invalid root, error stack:
PMPI_Bcast(451): MPI_Bcast(buf=0x7ffe446fe808, count=1, MPI_INTEGER, root=-1073741822, comm=comm=0x84000003) failed
PMPI_Bcast(412): Invalid root (value given was -1073741822)
Abort(738795271) on node 1 (rank 1 in comm 0): Fatal error in PMPI_Bcast: Invalid root, error stack:
PMPI_Bcast(451): MPI_Bcast(buf=0x7ffe6bbeb588, count=1, MPI_INTEGER, root=-1073741822, comm=comm=0x84000002) failed
PMPI_Bcast(412): Invalid root (value given was -1073741822)

**Comment by scivision** [comment]
I think Scotch and MUMPS have to both be built with 64-bit or 32-bit. I updated the build script for Scotch to have this option.

So pick a new install test directory, say ~/mumps64 and do like the following from the mumps/ Git source directory:

```sh
cmake -Dintsize64=on -Dprefix=~/mumps64 -P scripts/build_scotch.cmake

cmake -Bbuild -DCMAKE_PREFIX_PATH=~/mumps64 -Dintsize64=on -Dscotch=on 
cmake --build build
```

Then link your project as you did above.

**Reply by GuoqiMa** [reply]
@scivision  Thanks for you reply, my MUMPS now does include ZMUMPS, so this is not for 64-bit integer version, I have to build another one?  and also for all others

**Reply by GuoqiMa** [reply]
@scivision Since I only use METIS now, do I still need to rebuild 64-bit SCOTCH, and others like PORD, etc. can I remove them while linking?

**Comment by scivision** [comment]
I separated out METIS and Scotch so they can be used independently like the upstream MUMPS intended. I would do a fresh build using MUMPS v5.5.1.9 or newer:

```sh
cmake -Dintsize64=on -Dprefix=~/mumps64 -P scripts/build_metis.cmake

cmake -Bbuild -DBUILD_COMPLEX16=on -DCMAKE_PREFIX_PATH=~/mumps64 -Dintsize64=on -Dmetis=on 
cmake --build build
```
